bloomberg/bucklescript
# React jsx ppx: split ppx into two files again

Based on https://github.com/facebook/reason/pull/1291, which shrinks the size of the ppxes from 70k to 20k lines. The remaining's all omp I think.

Test:

- clone https://github.com/chenglou/reason-react-example
- change package.json to have `"reason-react": "https://github.com/SanderSpies/react-on-reason.git"` in `dependencies` and `"bs-platform": "https://github.com/chenglou/bucklescript.git#50b789823e91e5477965ba0227ce7c5d98eae7ec"` in `devDependencies` (points to this PR).
- `npm install && npm start`

See no significant artifact changes and no compilation error.

Then:

- Change `build.ninja` to use `reason_react_jsx = -ppx 'path/to/reason-react-example/node_modules/bs-platform/bin/reactjs_jsx_ppx_2.exe'`
- `./node_modules/bs-platform/bin/ninja.exe  -C lib/bs`

See compilation errors.
